### PR TITLE
Add Bruggeman EMA and depolarisation_factor property

### PR DESCRIPTION
**Pull request by haydenrob**

### CODE REVIEW

**Comment by haydenrob** [comment]
@andyfaff all tests in test_refellips.py passed locally for me, but test 11 fails on test_linux - I'm not sure why. Perhaps the way I've implemented warnings isn't correct for the Bruggeman approach?

File: refellips/structureSE.py [hunk 1]
```diff
@@ -557,6 +565,20 @@ def solvent(self, ri):
...
             self._solvent = solv
 
+    @property
+    def depolarisation_factor(self):
+        """ """
+        return self._depolarisation_factor
+
+    @depolarisation_factor.setter
+    def depolarisation_factor(self, value):
+        if not isinstance(value, (int, float)):
```
Comment (andyfaff) [inline refellips/structureSE.py:575]:
```suggestion
        if 0 <= float(value) <= 1:
            self._depolarisation_factor = 1
        else:
             raise ValueError("Depolarisation factor needs to be within [0, 1].")
```

File: refellips/structureSE.py [hunk 2]
```diff
@@ -605,7 +627,33 @@ def slabs(self, **kwds):
...
+        >>> StructureSE.ema = 'linear'
+        e_linear = e_h * (1 - vf) + e_i * vf
+
+        >>>StructureSE.ema = 'maxwell-garnett'
+        >>>StructureSE.depolarisation_factor = 1/3
+        top = e_h + (depolarisation_factor * (1 - vf) + vf) * (e_i - e_h)
+        bottom = e_h + depolarisation_factor * (1 - vf) * (e_i - e_h)
+        e_MG = e_h * top_r / bottom_r
+
+        >>>StructureSE.ema = 'bruggeman'
```
Comment (andyfaff) [inline refellips/structureSE.py:648]: Space between >>> and S.

**Comment by andyfaff** [comment]
Can you provide me with the equations/paper where the Bruggeman EMA is given? I'm not sure that you've implemented it correctly.

A clue to why that test is failing can be found [here](https://github.com/refnx/refellips/runs/5638613964?check_suite_focus=true#step:5:162). The sqrt calculation on `refellips/structureSE.py:1014` is being provided with an invalid value.
I then changed the code to `print(slabs[..., 1:3])` just before the sqrt. It turns out that all the values in column 1 are negative. Pretty sure that refractive indices should be positive. Anyway, the sqrt operation is trying to take the sqrt of a negative number and it doesn't like it.

Which brings me to a further point, are we doing any of the EMA calculations correctly?? Let's consider the `'linear'` EMA. The first thing we do is to square `n` and `k` separately:
```
>>> n = np.array([1, 2])
>>> n = n**2
>>> print(n)
[1 4]
```
However, this is totally different to:
```
>>> np.power(complex(1, 2), 2)
(-3+4j)
```

The averaging of the solvent and material is then done separately on `n**2` (column 1) and `k**2` (column 2), before taking the sqrt.

*Should we be performing these calculations using the first or second forms?*

I would hope that the test suite should detect this.

**Comment by haydenrob** [comment]
Just for reference, both EMA calcs are drawn from [this paper](https://opg.optica.org/josaa/fulltext.cfm?uri=josaa-33-7-1244&id=344272).

- Maxwell-Garnett comes from Eq 34
- Bruggeman EMA comes from Eq 39

**Comment by haydenrob** [comment]
@andyfaff okay to merge?

**Comment by andyfaff** [comment]
It looks ok to me. I think a little refactor might be in order in a later PR. I'm thinking about introducing a `MixedMaterialSlabSE` class, and it would be nice to use the overall_RI function to do that.

**Comment by andyfaff** [comment]
I haven't checked the maths, relying on you to do that.

**Comment by andyfaff** [comment]
If you're happy then merge.

**Comment by haydenrob** [comment]
I like the idea of the `MixedMaterialSlabSE`, but definitely something to look at later. Maths should be all good, and it converges with WVASE calcs as well.

**Pull request merged**